src/btree/bt_cursor.c

```diff
@@ -927,6 +937,58 @@ __btcur_search_neighboring(WT_CURSOR_BTREE *cbt, WT_CURFILE_STATE *state, int *e
         if (*exact <= 0)
             return (ret);
     }
+
+    return (ret);
+}
+
+/*
+ * __btcur_search_neighboring --
+ *     Search for a valid record around the cursor location.
+ *
+ */
+static int
+__btcur_search_neighboring(
+  WT_CURSOR_BTREE *cbt, WT_CURFILE_STATE *state, bool prepare_conflict, int *exact)
+{
+    WT_DECL_RET;
+
```

Comment:
Is there a reason we removed the `WT_ASSERT(!prepare_conflict || (prepare_conflict && compare != 0))` check? I think it's still useful to have. 

Alternately we could change the `else` branch below to
```
else if(cbt->compare > 0) {
    ret = __btcur_search_walk_prev(cbt, state, exact);
} else {
    // Unreachable. We should never call this function when we find a prepared update 
    // i.e. (prepare_conflict && compare == 0)
    WT_ASSERT(session, false);
}
```

Reply:
The previous assert is redundant now. I still prefer to assert WT_ASSERT(CUR2S(cbt), cbt->compare > 0); as currently it is. I feel it is weird to have WT_ASSERT(session, false); in the code. If we go that direction, we should return an error instead of an assert.

Reply:
Ah yeah good point. Happy to leave this as is 👍